lib/Runtime/Library/JavascriptGenerator.cpp

```diff
@@ -247,6 +262,419 @@ namespace Js
+    }
+
+    Var JavascriptGenerator::EntryAsyncGeneratorAwaitYield(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        ScriptContext* scriptContext = function->GetScriptContext();
+        PROBE_STACK(function->GetScriptContext(), Js::Constants::MinStackDefault);
+        ARGUMENTS(args, callInfo);
+        Assert(!(callInfo.Flags & CallFlags_New));
+        AssertOrFailFastMsg(args.Info.Count > 1, "Should never call EntryAsyncGeneratorAwaitYield without a parameter");
+        AsyncGeneratorNextProcessor* resumeNextReturnProcessor = VarTo<AsyncGeneratorNextProcessor>(function);
+        JavascriptGenerator* generator = resumeNextReturnProcessor->GetGenerator();
+        generator->SetState(GeneratorState::Suspended);
+
+        generator->AsyncGeneratorResolve(args[1], false);
+        return scriptContext->GetLibrary()->GetUndefined();
+    }
+
+    Var JavascriptGenerator::EntryAsyncGeneratorAwaitYieldStar(RecyclableObject* function, CallInfo callInfo, ...)
+    {
+        ScriptContext* scriptContext = function->GetScriptContext();
+        PROBE_STACK(function->GetScriptContext(), Js::Constants::MinStackDefault);
+        ARGUMENTS(args, callInfo);
+        Assert(!(callInfo.Flags & CallFlags_New));
+        AssertOrFailFastMsg(args.Info.Count > 1, "Should never call EntryAsyncGeneratorAwaitYield without a parameter");
+        AsyncGeneratorNextProcessor* resumeNextReturnProcessor = VarTo<AsyncGeneratorNextProcessor>(function);
+        JavascriptGenerator* generator = resumeNextReturnProcessor->GetGenerator();
+        generator->SetState(GeneratorState::Suspended);
+
+        if (VarIs<RecyclableObject>(args[1]))
+        {
+            RecyclableObject* yieldData = UnsafeVarTo<RecyclableObject>(args[1]);
+            Var value = JavascriptOperators::GetProperty(yieldData, PropertyIds::value, scriptContext);
+
+            JavascriptOperators::OP_AsyncYield(generator, value, scriptContext);
+            return scriptContext->GetLibrary()->GetUndefined();
+        }
+        Var error = generator->CreateTypeError(JSERR_NonObjectFromIterable, scriptContext, _u("yield*"));
+        JavascriptOperators::OP_AsyncYield(generator, error, scriptContext);
+        return scriptContext->GetLibrary()->GetUndefined();
+    }
+
+    Var JavascriptGenerator::AsyncGeneratorEnqueue(Var thisValue, ScriptContext* scriptContext, Var input, JavascriptExceptionObject* exceptionObj, const char16* apiNameForErrorMessage)
+    {
+        JavascriptPromise* promise = JavascriptPromise::CreateEnginePromise(scriptContext);
+
+        if (!VarIs<JavascriptGenerator>(thisValue))
+        {
+            Var error = CreateTypeError(JSERR_NeedObjectOfType, scriptContext, apiNameForErrorMessage, _u("AsyncGenerator"));
+            promise->Reject(error, scriptContext);
+            return promise;
+        }
+
+        JavascriptGenerator* generator = UnsafeVarTo<JavascriptGenerator>(thisValue);
+
+        if (!generator->GetIsAsync())
```

Comment:
(Same question as above.)

Reply:
Yes similar to point above though this one would be specifically if you did:
```js
function* genF(){}
async function* asyncGenF(){}
const nextMethod = asyncGenF().next;
nextMethod.call(genF());
```